service-workers/service-worker/iframe-controller.https.html

```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<title>Service worker inheritance</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+
+promise_test(function(t) {
+    var script = 'iframe-controller-worker.https.js';
+    var scope1 = 'iframe-controller.https.html';
+    var scope2 = 'resources/blank.html';
+    var script1 = script + '?for_scope1';
+    var script2 = script + '?for_scope2';
+    var registration1, registration2;
+    var parent_controller, child_controller;
+    return service_worker_unregister_and_register(t, script1, scope1)
+      .then(function(r) {
+          registration1 = r;
+          return wait_for_state(t, registration1.installing, 'activated');
+        })
+      .then(function() {
+          return service_worker_unregister_and_register(t, script2, scope2);
+        })
+      .then(function(r) {
+          registration2 = r;
+          return wait_for_state(t, registration2.installing, 'activated');
+        })
+      .then(function() {
+          // This client's active service worker is set to registration1's
+          // active worker by clients.claim() during the activation.
+          parent_controller = navigator.serviceWorker.controller;
+          return with_iframe('');
+        })
+      .then(function(f) {
+          child_controller = f.contentWindow.navigator.serviceWorker.controller;
+          assert_equals(child_controller.scriptURL, parent_controller.scriptURL,
+                        'about:blank Document should inherit parent\'s SW.');
+        })
+      .then(function() {
+          return with_srcdoc_iframe('<p>Some HTML</p>');
```

Comment:
This is a good point indeed. I addressed them as suggested basically but put the registration and activation chain in the first promise_test as it needs |t|.